### PR TITLE
Validations and tests

### PR DESCRIPTION
**Pull request by Daphineatim**
### In this milestone, the following changes were implemented;

- [ ] Added the following validations:

For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

- [ ] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  it 'Update like counter for post' do
```
Comment (MahmoudBakr23) [inline spec/models/like_spec.rb:8]: - [x] Well-written tests so far for your models' validations. However, you've missed a small technical point to implement, and it is the `describe` method for describing each test case as a unique entity in order to make it easier for the developers to read and track the code.
see an example!
```ruby
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```
_Please implement the changes in all `_spec` files_

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Daphineatim 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done installing the RSpec gem
- Great job specifying the tests for validating your models
- Descriptive PR and README file
- Correct Linters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/like_spec.rb (hunk 1 above)
Comment (Daphineatim) [reply spec/models/like_spec.rb:8]: Thank you so much, I have made the changes.

**Review by sumairq — APPROVED** [review]
### Hi @Daphineatim , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**